src/promptflow/promptflow/_core/flow_execution_context.py

```diff
@@ -141,6 +142,14 @@ async def invoke_tool_async(self, node: Node, f: Callable, kwargs):
             self._run_tracker.end_run(node_run_id, result=result, traces=traces)
             flow_logger.info(f"Node {node.name} completes.")
             return result
+        # User tool should reraise the CancelledError after its own handling logic,
+        # so that the error can propagate to the scheduler for handling.
+        # Otherwise, the node would end with Completed status.
+        except asyncio.CancelledError as e:
+            logger.info(f"Node {node.name} in line {self._line_number} is cancelled.")
+            traces = Tracer.end_tracing(node_run_id)
+            self._run_tracker.end_run(node_run_id, ex=e, traces=traces)
+            raise
```

Comment:
Reraise asyncio.CancelledError will break single node test.
And runtime will also receive this exception.
Maybe we should keep `raise_ex`'s function for BaseException?
![image](https://github.com/microsoft/promptflow/assets/17527303/f5ad35df-4d73-45b7-b16c-686c6d3c454e)

Reply:
It's by design.
We depend on these BaseExceptions to stop whole process.

Reply:
`raise_ex` should not apply for BaseException. And persist logic is triggered in the "finally block".